packages/jest-cli/src/lib/init/generate_config_file.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {defaults, descriptions} from 'jest-config';
+
+const generateConfigFile = (results: {[string]: boolean}): string => {
+  const {typescript, coverage, clearMocks, environment} = results;
+
+  const overrides: Object = {};
+
+  if (typescript) {
+    Object.assign(overrides, {
+      globals: {
+        'ts-jest': {
+          tsConfigFile: 'tsconfig.json',
+        },
+      },
+      moduleFileExtensions: ['ts', 'tsx', 'js'],
+      testMatch: ['**/__tests__/*.+(ts|tsx|js)'],
+      transform: {
+        '^.+\\.(ts|tsx)$': 'ts-jest',
+      },
+    });
+  }
+
+  if (coverage) {
+    Object.assign(overrides, {
+      coverageDirectory: 'coverage',
+    });
+  }
+
+  if (environment === 'node') {
+    Object.assign(overrides, {
+      testEnvironment: 'node',
+    });
+  }
+
+  if (clearMocks) {
+    Object.assign(overrides, {
+      clearMocks: true,
+    });
+  }
+
+  const overrideKeys: Array<string> = Object.keys(overrides);
+
+  const stringifyOption = (
```

Comment:
does this have to exist inside the `generateConfigFile` closure?

Reply:
I'll move it out of there 😉